src/syrupy/report.py

```diff
@@ -346,9 +346,14 @@ def _provided_nodes_match_name(
         self, snapshot_name: str, provided_nodes: List[List[str]]
     ) -> bool:
         """
-        Check that a snapshot name matches the node paths provided
+        Check that a snapshot name matches the node paths provided.
+        If no nodes are filtered, provided_nodes is empty, which means
+        all nodes should be matched.
```

Comment:
@iamogbz Is this statement correct?

Reply:
Just added another test which is failing, so this is definitely not true. We need to somehow filter provided_nodes so it's only considering relevant nodes. The problem is that the nodes are not fully qualified so they're ambiguous between files.